python/ray/data/_internal/stats.py

```diff
@@ -266,6 +267,51 @@ def __init__(self, max_stats=1000):
             tag_keys=iter_tag_keys,
         )
 
+        # === Dataset and Operator Metadata Metrics ===
+        dataset_tags = ("dataset",)
+        self.dataset_progress = Gauge(
+            "data_dataset_progress",
+            description="Progress of dataset execution",
+            tag_keys=dataset_tags,
+        )
+        self.dataset_total = Gauge(
+            "data_dataset_total",
+            description="Total work units in blocks for dataset",
+            tag_keys=dataset_tags,
+        )
+        self.dataset_total_rows = Gauge(
+            "data_dataset_total_rows",
+            description="Total work units in rows for dataset",
+            tag_keys=dataset_tags,
+        )
+        self.dataset_state = Gauge(
+            "data_dataset_state",
+            description=f"State of dataset ({', '.join([f'{s.value}={s.name}' for s in DatasetState])})",
+            tag_keys=("dataset",),
+        )
+
+        operator_tags = ("dataset", "operator")
+        self.operator_progress = Gauge(
+            "data_operator_progress",
+            description="Progress of operator execution",
+            tag_keys=operator_tags,
+        )
+        self.operator_total = Gauge(
+            "data_operator_total",
```

Comment:
```suggestion
            "data_operator_estimated_total_blocks",
```
This is the estimated total number of blocks right?

Reply:
Same for other "total" metrics. let's add "estimated" to avoid confusion.